september/src/main/java/edu/asu/diging/gilesecosystem/september/core/db/impl/MessageDbClient.java

```diff
@@ -18,73 +20,107 @@
-        query = query.select(root);      
-        
-        return em.createQuery(query).getResultList();
-    }
-    
-    @Override
-    public List<Message> getMessages(int offset, int pageSize, String sortField) {
-        CriteriaBuilder builder = em.getCriteriaBuilder();
-        CriteriaQuery<Message> query = builder.createQuery(Message.class);
-        Root<Message> root = query.from(Message.class);
-        query = query.select(root).orderBy(builder.desc(root.get(sortField)));    
-        
-        TypedQuery<Message> finalQuery = em.createQuery(query);
-        finalQuery.setFirstResult(offset).setMaxResults(pageSize);
-        return finalQuery.getResultList();
-    }
-    
-    @Override
-    public int getNumberOfMessages() {
-        CriteriaBuilder builder = em.getCriteriaBuilder();
-        CriteriaQuery<Message> query = builder.createQuery(Message.class);
-        Root<Message> root = query.from(Message.class);
-        query = query.select(root);      
-        
-        return em.createQuery(query).getResultList().size();
-    }
-    
-    @PreDestroy
-    public void shutdown() {
-        em.close();
-        em = null;
-        
-        try {
-            DriverManager.getConnection("jdbc:derby:;shutdown=true");
-        } catch (SQLException e) {
-            logger.error("Derby is shutdown.", e);
-        }
-    }
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	@PersistenceContext(unitName = "DataPU")
+	private EntityManager em;
+
+	@Override
+	protected String getIdPrefix() {
+		return "MSG";
+	}
+
+	@Override
+	protected IMessage getById(String id) {
+		return em.find(Message.class, id);
+	}
+
+	@Override
+	protected EntityManager getClient() {
+		return em;
+	}
+
+	@Override
+	public List<Message> getMessages() {
+		CriteriaBuilder builder = em.getCriteriaBuilder();
+		CriteriaQuery<Message> query = builder.createQuery(Message.class);
+		Root<Message> root = query.from(Message.class);
+		query = query.select(root);
+
+		return em.createQuery(query).getResultList();
+	}
+
+	@Override
+	public List<Message> getMessages(int offset, int pageSize, String sortField) {
+		CriteriaBuilder builder = em.getCriteriaBuilder();
+		CriteriaQuery<Message> query = builder.createQuery(Message.class);
+		Root<Message> root = query.from(Message.class);
+		query = query.select(root).orderBy(builder.desc(root.get(sortField)));
+
+		TypedQuery<Message> finalQuery = em.createQuery(query);
+		finalQuery.setFirstResult(offset).setMaxResults(pageSize);
+		return finalQuery.getResultList();
+	}
+
+	public List<MessageType> filterStringtoList(String regex) {
+		List<MessageType> filter = new ArrayList<MessageType>();
+		StringTokenizer st = new StringTokenizer(regex, "|");
+		while (st.hasMoreTokens()) {
+			filter.add(MessageType.valueOf(st.nextToken()));
+		}
+		return filter;
+
+	}
+
+	@Override
+	public List<Message> getFilteredMessages(int offset, int pageSize, String regex, String sortField) {
```

Comment:
this method should get a list of types that should be filtered, there should be no string parsing going on here. This is not the DB connector's job.